### PR TITLE
✨ Add `report_url` param to `CloudForCustomers.extract_records()`

### PR DESCRIPTION
**Pull request by Rafalz13**
## Summary
Added report_url argument.

## Importance
Deployments are not working without this argument

## Checklist

This PR:
- [ ] follows the guidelines laid out in `CONTRIBUTING.md`
- [ ] links relevant issue(s)
- [ ] adds/updates tests (if appropriate)
- [ ] adds/updates docstrings (if appropriate)
- [ ] adds an entry in `CHANGELOG.md`

### CODE REVIEW

File: src/viadot/sources/cloud_for_customers.py [hunk 1]
```diff
@@ -284,7 +284,7 @@ def get_response(
         return handle_api_response(
             url=url,
             params=filter_params,
-            auth=(username, password),
+            auth=(username, password._secret_value),
```
Comment (trymzet) [inline src/viadot/sources/cloud_for_customers.py:287]: The way to do this in pydantic is to use `get_secret_value()` https://docs.pydantic.dev/latest/examples/secrets/

File: src/viadot/sources/cloud_for_customers.py [hunk 2]
```diff
@@ -318,7 +318,7 @@ def to_df(
         tests: dict[str, Any] = kwargs.get("tests", {})
 
         url = url or self.url
-        records = self.extract_records(url=url)
+        records = self.extract_records(url=url, report_url=self.report_url)
```
Comment (trymzet) [inline src/viadot/sources/cloud_for_customers.py:321]: Why use different logic than for `url`? Would be simpler if both are treated the same way.

**Review by trymzet — CHANGES_REQUESTED** [review]
Added feedback. Pls also use standard naming convention for PR title (see https://github.com/dyvenia/viadot/blob/2.0/CONTRIBUTING.md#commit-messages-style).

File: src/viadot/sources/cloud_for_customers.py (hunk 2 above)
Comment (Rafalz13) [reply src/viadot/sources/cloud_for_customers.py:321]: @fdelgadodyvenia could you explain?

Comment (Rafalz13) [reply src/viadot/sources/cloud_for_customers.py:321]: Added issue : https://github.com/dyvenia/viadot/issues/1081

**Review by trymzet — APPROVED** [review]

**Review by trymzet — APPROVED** [review]

**Pull request merged**